src/source/src/manager.rs

```diff
@@ -32,20 +32,19 @@ use crate::monitor::SourceMetrics;
 use crate::table::TableSource;
 use crate::{ConnectorSource, SourceFormat, SourceImpl, SourceParserImpl};
 
-pub type SourceRef = Arc<SourceImpl>;
+pub type SourceDescRef = Arc<SourceDesc>;
+type WeakSourceDescRef = Weak<SourceDesc>;
 
 /// The local source manager on the compute node.
 #[async_trait]
-pub trait SourceManager: Debug + Sync + Send {
-    fn get_source(&self, source_id: &TableId) -> Result<SourceDesc>;
-    fn drop_source(&self, source_id: &TableId) -> Result<()>;
-
-    /// Clear sources, this is used when failover happens.
-    fn clear_sources(&self) -> Result<()>;
+pub trait TableSourceManager: Debug + Sync + Send {
```

Comment:
Let's remove the trait and directly name `struct MemSourceManager` as `struct TableSourceManager`? cc @yezizp2012

Reply:
This is also something confusing me that the trait is only implemented once..
Absolutely we can remove the trait and use struct directly : ) I will do this replacement.